distributed/worker.py

```diff
@@ -2233,6 +2238,47 @@ def _get_client(self, timeout=3):
                 assert self._client.status == 'running'
         return self._client
 
+    @gen.coroutine
+    def register_worker_environments(self, stream=None, environments=None):
+        result = {}
+        for name, env in environments.items():
+            if name not in self.environments:
+                try:
+                    env = pickle.loads(env)
+                    meets_condition = yield self.executor_submit('condition', env.condition)
+                    # TODO: Should this be in some kind of "while running" loop with
+                    # a timeout?
+                    if gen.is_future(meets_condition):
+                        meets_condition = meets_condition.result()
+                    if meets_condition:
+                        # TODO: This currently blocks. I expect that that is
+                        # not desirable, as a long setup task may look like a
+                        # worker that's become unresponsive.
```

Comment:
This doesn't seem to block anymore.  I suspect that this comment is out of date.